### DISCUSSION TITLE
Cody - Your Competitive Programming Friend

### DISCUSSION THREAD

**Discussion by pri1311** [scaleracademy/open-source-projects#8] [Ideas]
## Project Summary
Cody is a VSCode extension that alerts the users about upcoming/current coding contests on various competitive programming websites like CodeChef, Codeforces, Atcoder, etc.

It should download/display problem statements as well as test cases and submission results to the users in the editor itself. Users should also be able to submit their code/solution from VS Code itself instead of having to manually submit it on the contest website.

In addition, if possible instead of just having contest questions loaded and displayed, we can also let users view and solve problems from practice sections of various websites like Leetcode, GFG, etc.

## Features
* Alerts for coding contests.
* Let users set reminders manually.
* Give the user the choice of shutting alerts for specific websites.
* Display problem statement along with test cases on the editor itself.
* Submit solution from editor and show results.
* Keyboard shortcuts for various tasks like showing upcoming contests, going to the next problem, submitting the code, etc.
* Allow users to view and submit practice problems as well.
* Show a dashboard of how many problems solved, number of contests, etc.

**Comment by championswimmer** [comment]
Are there any APIs you are aware of these websites from where we can easily fetch details of upcoming competitions ? 

Also was thinking maybe we can add a social element where it shows if your friends are participating in some existing contests.

**Reply by pri1311** [reply]
I am still researching for that, but I know that some websites like Codeforces and hackerrank have their own APIs.

Also, the friends feature would be really cool

**Reply by pri1311** [reply]
https://kontests.net/api this is one API, but it returns very basic information

**Reply by revantg** [reply]
Check this [API](http://contesttrackerapi.herokuapp.com/). Might be useful.
Also in case you haven't heard, [Clist](https://clist.by/) is a good website for checking upcoming, running contests. It's response has most of the relevant info for all contests.

**Reply by pri1311** [reply]
Another thing that comes to mind, is code snippets that we can provide, which are generally used in cp

**Comment by TAbhay** [comment]
@pri1311 

Website :  _[Stopstalk](https://www.stopstalk.com/)_
Hey , It's not a extension but it contains all the features as you mentioned [](url)
a) User can add various coding platform profile ( Codeforces , UvaOnline judge ....)
b) Upcoming , past and present contents
c) Solved and unsolved questions on different platforms , trending problems , recommended problems
d) Add friends
e) Looking for Job ... it can also help you in this 
_The thing which I like most_ 
**You can also stalk your college friends on it**
